app/scripts/update-translation.js

```diff
@@ -0,0 +1,98 @@
+import fs from 'node:fs/promises'
+import path from 'node:path'
+import { fileURLToPath } from 'url';
+import { dirname } from 'path';
+import { generateText } from 'ai';
+import { openai } from '@ai-sdk/openai';
+
+// Convert the module's URL to a file path
+const __filename = fileURLToPath(import.meta.url);
+const __dirname = dirname(__filename);
+
+if (!process.env.OPENAI_API_KEY) {
+  console.error('Please set the OPENAI_API_KEY environment variable')
+  process.exit(1)
+}
+
+const model = openai('gpt-4');
+
+function stripQuotes(s) {
+  return s.replace(/^"(.*)"$/, '$1')
+}
+
+async function translateString(sourceLanguage, targetLanguage, key, sourceValue) {
```

Comment:
@mircorudolph this section might be interesting to you. I think we could probably add more context here for related strings. Any suggestions also welcome.

Reply:
The way I see it we pass in a single file and a single string to translate at a time?
For this I do not have a good idea how to add extra context easily the same way I did in the global-data translation where all the sentences are given in a single dictionary per sentence.
We could maybe load the same file for all the available locales and add it into the context as additional information to translate, instead of choosing a single source language ('en' in this case). But that would require some more code changes and we could check first if the translation is good enough and see if its necessary. 
Translations should work pretty well for zero-shot.

Reply:
Awesome. I thought providing some other context would help keep the vocabulary consistent, like, which synonym we choose to translate the same word. Anyway, thank you!